htdocs/societe/class/api_thirdparties.class.php

```diff
@@ -1170,6 +1171,127 @@ function deleteCompanyBankAccount($id, $socid){
 		return $account->delete(DolibarrApiAccess::$user);
 	}
 
+	/**
+	 * Generate a sepamandate Document
+	 * 
+	 * @param int $socid thirdparty id
+	 * 
+	 * @return array Check success
+	 * 
+	 * @url GET /generateMandat/{socid}
+	 */
+	public function generateMandat($socid){
+
+		$this->langs->load("database");
+		$this->langs->load("main");
+		$this->langs->load("dict");
+		$this->langs->load("agenda");
+		$this->langs->load("margins");
+		$this->langs->load("resource");
+		$this->langs->load("commercial");
+		$this->langs->load("ecm");
+		$this->langs->load("products");
+		$this->langs->load("companies");
+		$this->langs->load("banks");
+		$this->langs->load("bills");
+		$this->langs->load("withdrawals");
+
+
+		$model = "sepamandate";
+		
+		$this->company->fetch($socid);
+		
+		$action = 'builddoc';
+		if(! DolibarrApiAccess::$user->rights->societe->creer)
+			throw new RestException(401);
+		
+		
+		// Reload to get all modified line records and be ready for hooks
+		
+		
+		$this->company->setDocModel($user, $model);
+		
+		$this->company->fk_bank = $this->company->fk_account;
+		
+		$outputlangs = $this->langs;
+		$newlang='';
+		
+		
+		
+		if ($this->conf->global->MAIN_MULTILANGS && empty($newlang) && GETPOST('lang_id','aZ09')) $newlang=GETPOST('lang_id','aZ09');
+		if ($this->conf->global->MAIN_MULTILANGS && empty($newlang) && isset($this->company->thirdparty->default_lang)) $newlang=$this->company->thirdparty->default_lang;  // for proposal, order, invoice, ...
+		if ($this->conf->global->MAIN_MULTILANGS && empty($newlang) && isset($this->company->default_lang)) $newlang=$this->company->default_lang;                  // for thirdparty
+		if (! empty($newlang))
+		{
+			$outputlangs = new Translate("",$conf);
+			$outputlangs->setDefaultLang($newlang);
+		}
+		
+		// To be sure vars is defined
+		if (empty($hidedetails)) $hidedetails=0;
+		if (empty($hidedesc)) $hidedesc=0;
+		if (empty($hideref)) $hideref=0;
+		if (empty($moreparams)) $moreparams=null;
+		
+		
+		$sql = "SELECT rowid";
+		$sql.= " FROM ".MAIN_DB_PREFIX."societe_rib";
+		if ($socid) $sql.= " WHERE fk_soc  = ".$socid." ";
```

Comment:
A thirdparty can have several bank account and also this table contains also credit card, not only bank account.
So you should introduce a parameter "companybankid" that is id in table societe_rib as a parameter of method generateMandat and use it as the value of $accounts[0]->id get later (instead of getting the first record found that may not be the correct one).